src/vm/ast/sugar_expander/mod.rs

```diff
@@ -0,0 +1,193 @@
+use std::convert::TryInto;
+use vm::representations::{PreSymbolicExpression, PreSymbolicExpressionType, SymbolicExpression, SymbolicExpressionType};
+use vm::types::{QualifiedContractIdentifier, Value, PrincipalData, StandardPrincipalData};
+use vm::ast::types::{ContractAST, BuildASTPass};
+use vm::ast::errors::{ParseResult, ParseError, ParseErrors};
+use vm::functions::NativeFunctions;
+
+pub struct SugarExpander {
+    issuer: StandardPrincipalData   
+}
+
+impl BuildASTPass for SugarExpander {
+
+    fn run_pass(contract_ast: &mut ContractAST) -> ParseResult<()> {
+        let pass = SugarExpander::new(contract_ast.contract_identifier.issuer.clone());
+        let expressions = pass.run(&mut contract_ast.pre_expressions);
+        contract_ast.expressions = expressions;
+        Ok(())
+    }
+}
+
+impl SugarExpander {
```

Comment:
The SugarExpander could eliminate a lot of clones by being destructive, that is, `transform` would have the following signature:

```
pub fn transform(&self, pre_expressions: &mut Vec<PreSymbolicExpression>)
```

And then you could use `drain()` rather than `iter()`. I think you'd need to also use a Vec<> in the `PreSymbolicExpressionType::List`.

Reply:
Hm ok, I'll put myself in the arena with the borrow checker

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/1633a7611ecafbf5393e06092c01e3c8265834c6